pkg/server/sotw/v2/server.go

```diff
@@ -153,7 +107,11 @@ func (s *server) process(stream Stream, reqCh <-chan *discovery.DiscoveryRequest
 
 	// a collection of stack allocated watches per request type
 	var values watches
-	values.Init()
+	bufferSize := 1
+	if defaultTypeURL == resource.AnyType {
+		bufferSize = 8
```

Comment:
Can you bring configurability for this? xDS is constantly evolving so it's not going to stay at 8 for long.

Reply:
I think we need 2 sizes here to be configurable - for ADS and for non-ADS separately, so that people who use both ADS and non-ADS within same process don't use big buffers while they are not needed. WDYT?

Reply:
Sounds fine to me. "1" might be OK for non-ADS. ADS might have more than 8 types multiplexed.

Reply:
At Lyft, we use non ADS with buffer size 1 , but we sometimes experiment with other sizes. We would need configurability for the buffer size.